src/systemctl/systemctl-edit.c

```diff
@@ -427,11 +427,23 @@ static int find_paths_to_edit(
         _cleanup_(edit_file_free_all) EditFile *edit_files = NULL;
         _cleanup_(lookup_paths_free) LookupPaths lp = {};
         size_t n_edit_files = 0;
+        const char *dropin, *suffix;
         int r;
 
         assert(names);
         assert(ret_edit_files);
 
+        if (!arg_edit_filename)
+                dropin = strdupa_safe("override.conf");
+        else if (!endswith(arg_edit_filename, ".conf"))
+                dropin = strjoina(arg_edit_filename, ".conf");
```

Comment:
Please use a normal allocation for this. The argument is external and we shouldn't use a stack allocation for something of unknown size.

Reply:
> I like the idea, but not the option name... (unneeded long) I mean why the `-edit` part when the command already is` systemctl edit` ?.... I would personally prefer `--drop-in=` instead, which is IMHO more to the point.

I agree. `--drop-in=` sounds OK. Maybe add an (undocumented) alias `--dropin=`, because people will forget the dash all the time.

> Plus if it's true that we already can edit multiple files using one edit command, how is this option being related to them individually?

I don't think `systemctl edit` allows more than one file to be edited at a time. The approach with just overriding one filename to operate on seems just fine.

Thus:
1. clean up the allocations
2. rename the option

Looks good otherwise.

Reply:
When we introduce an option, then I think we should not add an undocumented 'alias'. We only add such an alias if the option is in a wrong English. Though, I do not have strong opinion about whether `--drop-in` or `--dropin`.

Reply:
> Please use a normal allocation for this.

```c
if (!arg_dropin || streq(arg_dropin, ""))
        dropin = "override.conf";
else if (!endswith(arg_dropin, ".conf"))
        dropin = strjoin(arg_dropin, ".conf");
else
        dropin = arg_dropin;
```

How could I free the pointer only in the second case 🤔

Reply:
@yuwata ,
> Though, I do not have strong opinion about whether `--drop-in` or `--dropin`.

The functionality is refered-to as "drop-in" in all the documentation, so it's more correct to use that term IMHO :stuck_out_tongue_winking_eye: